api/meetings/serializers.py

```diff
@@ -0,0 +1,213 @@
+    def get_author_name(self, obj):
+        """
+        Returns the first bibliographic contributor's family_name if it exists.
+        Otherwise, return its fullname.
+        """
+        if getattr(obj, 'author_name', None):
+            # Field is annotated on queryset in ListView for filtering purposes
+            return obj.author_name
+        else:
+            author = self.get_author(obj)
+            if author:
+                return author.family_name if author.family_name else author.fullname
+            return None
+
+    def get_meeting_category(self, obj):
+        """
+        Returns the existance of a certain tag on the node.  If the first submission type tag exists,
+        return that.  Otherwise, return the second submission type tag as a default.
+        """
+        if getattr(obj, 'meeting_category', None):
+            # Field is annotated on queryset in ListView for filtering purposes
+            return obj.meeting_category
+        else:
+            meeting = self.context['meeting']
+            submission1_name = meeting.field_names.get('submission1')
+            submission2_name = meeting.field_names.get('submission2')
+            submission_tags = obj.tags.values_list('name', flat=True)
+            return submission1_name if submission1_name in submission_tags else submission2_name
+
+    def get_download_count(self, obj):
+        """
+        Return the download counts of the first osfstorage file
+        """
+        if getattr(obj, 'download_count', None):
+            return obj.download_count or 0
+        else:
+            node_ct = ContentType.objects.get_for_model(AbstractNode).id
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    """
+                    SELECT P.total
+                    FROM osf_basefilenode F, osf_pagecounter P
+                    WHERE (F.type = 'osf.osfstoragefile'
+                         AND F.provider = 'osfstorage'
+                         AND F.target_content_type_id = %s
+                         AND F.target_object_id = %s
+                         AND P._id = 'download:' || %s || ':' || F._id)
+                    ORDER BY F.id ASC
+                    LIMIT 1;
+                """, [node_ct, obj.id, obj._id],
+                )
+                result = cursor.fetchone()
+                if result:
+                    return int(result[0])
+                return 0
+
+    def get_download_link(self, obj):
+        """
+        First osfstoragefile on a node - if the node was created for a meeting,
+        assuming its first file is the meeting submission.
+        """
+        if getattr(obj, 'file_id', None):
+            submission_file = OsfStorageFile.objects.get(_id=obj.file_id)
+        else:
+            files = obj.files.order_by('created')
+            submission_file = files.first()
+
+        if submission_file:
+            return get_file_download_link(submission_file)
+        return None
```

Comment:
I've been putting restrictions on list views and not detail views.  So you could navigate directly to a conference submission with no file attached, but it would show up in the list view.

Reply:
Gotcha. I'll put in both of those requirements.